apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx

```diff
@@ -112,6 +115,25 @@ export const AiAssistantPanel = ({
 
   const { mutate: sendEvent } = useSendEventMutation()
   const { mutate: updateOrganization, isLoading: isUpdating } = useOrganizationUpdateMutation()
+  const [isApiKeySet, setIsApiKeySet] = useState<boolean>(true)
+
+  // check to see if the OPENAI_API_KEY env var is set in self-hosted
+  // so we can disable the chat editor and add a warning about manually adding the key
+  useEffect(() => {
+    if (IS_PLATFORM) return
+
+    const checkApiKey = async () => {
+      try {
+        const response = await fetch(`${BASE_PATH}/api/ai/sql/check-api-key`)
+        setIsApiKeySet(response.status === 200)
+      } catch (error) {
+        console.error('Error checking API key:', error)
+        setIsApiKeySet(false)
+      }
+    }
+
+    checkApiKey()
+  }, [])
 
```

Comment:
this could probably be a react query - will be cleaner i think

Reply:
also thinking - probably doesnt need to be self hosted specific? this check should also work for hosted too, probably a good safety net for us actually haha

Reply:
^ actually sorry think if we do this API call on the hosted dash it'll probably incur unnecessary functions cost